File: p2.cc
#include <iostream>

int main() {
	int sum = 0;
	int a = 0;
	int b = 1;
	while (a + b <= 4000000) {
		int temp = b;
		b += a;
		a = temp;
		if (b % 2 == 0) {
			sum += b;
		}
	}
	std::cout << sum << std::endl;
	return 0;
}

File: go/p2.go
package main

import "fmt"

func main() {
  i := 0
  j := 1
  x := 0
  for j <= 4000000 {
    if j % 2 == 0 {
      x += j
    }
    temp := j
    j += i
    i = temp
  }

  fmt.Println(x)
}